src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -62,6 +86,30 @@ internal void WriteAsUstar(Stream archiveStream, Span<byte> buffer)
             }
         }
 
+        // Asynchronously rites the current header as a Ustar entry into the archive stream and returns the value of the final checksum.
+        internal async Task<int> WriteAsUstarAsync(Stream archiveStream, Memory<byte> buffer, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            long actualLength = GetTotalDataBytesToWrite();
+            TarEntryType actualEntryType = GetCorrectTypeFlagForFormat(TarEntryFormat.Ustar);
+
+            int checksum = WritePosixNameMemory(buffer);
+            checksum += WriteCommonFields(buffer.Span, actualLength, actualEntryType);
+            checksum += WritePosixMagicAndVersion(buffer.Span);
+            checksum += WritePosixAndGnuSharedFields(buffer.Span);
+            int finalChecksum = WriteChecksum(checksum, buffer.Span);
```

Comment:
similar to the method above, can we reduce the code duplication between sync and async versions?